stan/math/rev/scal/fun/rising_factorial.hpp

```diff
@@ -8,28 +8,23 @@
 namespace stan {
   namespace math {
 
-    namespace {
+    /**
+     * Return tangent of the rising factorial function 
```

Comment:
This is just free-floating doc.  Every piece of doxygen doc should be attached to a class or a function or a variable.  I'm not sure where you meant this to go.  Also, it helps to be more specific than "argument" --- at the very least distinguish the first argument from the second argument.  

You don't need to document all the little helper classes. It's the functions that are going to be used by external clients that are important (though @syclik would probably argue everything should be documented).